pkg/sql/parser/encode.go

```diff
@@ -70,21 +77,34 @@ func encodeSQLStringWithFlags(buf *bytes.Buffer, in string, f FmtFlags) {
 	start := 0
 	escapedString := false
 	bareStrings := f.bareStrings
+	var quoteChar byte = '\''
+	if f.doubleQuoteStrings {
+		quoteChar = '"'
+	}
+	if mustQuoteStringsMap[strings.ToUpper(in)] {
```

Comment:
That's no good. 

1) I do not understand the special casing of the "NULL" string. It seems like this is the wrong place to check this.
2) You're mandating an allocation an scanning a potentially very large string just for two values that take 0 and 4 comparisons to check. Better have all 16 variants of "NULL" in the map and compare with `len(in) <= 4 && mqsm[in]`.